core/blockchain.go

```diff
@@ -340,10 +368,26 @@ func (self *BlockChain) Status() (td *big.Int, currentBlock common.Hash, genesis
 	return self.GetTd(self.currentBlock.Hash()), self.currentBlock.Hash(), self.genesisBlock.Hash()
 }
 
-func (self *BlockChain) SetProcessor(proc types.BlockProcessor) {
-	self.processor = proc
+// SetProcessor sets the processor required for making state modifications.
+func (self *BlockChain) SetProcessor(processor Processor) {
+	self.processor = processor
+}
+
+// SetValidator sets the validator which is used to validate incoming blocks.
+func (self *BlockChain) SetValidator(validator Validator) {
+	self.validator = validator
 }
 
+// Validator returns the current validator.
+func (self *BlockChain) Validator() Validator { return self.validator }
+
+// Processor returns the current processor.
+func (self *BlockChain) Processor() Processor { return self.processor }
+
+// AuxValidator returns the auxiliary validator (Proof of work atm)
+func (self *BlockChain) AuxValidator() pow.PoW { return self.pow }
```

Comment:
Not sure if this one can be changed dynamically, but if yes, then it also needs to be protected by the mutex.